--- a/Configurations/15-android.conf
+++ b/Configurations/15-android.conf
@@ -112,18 +112,17 @@
                 # *after* announcement that gcc is being phased out, so
                 # favouring clang is considered adequate. Those who insist
                 # have option to enforce test for gcc with CC=gcc.
-                if (which("$triarch-$cc") !~ m|^$ndk|) {
-                    die "no NDK $triarch-$cc on \$PATH";
+                if (realpath(which("$cc")) !~ m|^$ndk|) {
+                    die "no NDK $cc on \$PATH";
                 }
                 $user{CC} = $cc;
-                $user{CROSS_COMPILE} = "$triarch-";
+                $user{CROSS_COMPILE} = "";
             } elsif ($user{CC} eq "clang") {
                 die "no NDK clang on \$PATH";
             } else {
                 if (which("$triarch-gcc") !~ m|^$ndk/.*/prebuilt/([^/]+)/|) {
                     die "no NDK $triarch-gcc on \$PATH";
                 }
-                $cflags .= " -mandroid";
                 $user{CROSS_COMPILE} = "$triarch-";
             }
 
@@ -139,7 +138,7 @@
 
             $sysroot =~ s|^$ndk/||;
             $android_ndk = {
-                cflags   => "$cflags --sysroot=\$($ndk_var)/$sysroot",
+                cflags   => $cflags,
                 cppflags => $cppflags,
                 bn_ops   => $arch =~ m/64$/ ? "SIXTY_FOUR_BIT_LONG"
                                             : "BN_LLONG",

