--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -312,6 +312,7 @@
     )
 endif (WITH_SYMBOL_VERSIONING AND HAVE_LD_VERSION_SCRIPT AND ABIMAP_FOUND)
 
+if(OFF)
 add_library(${LIBSSH_SHARED_LIBRARY} SHARED ${libssh_SRCS})
 target_compile_options(${LIBSSH_SHARED_LIBRARY}
                        PRIVATE
@@ -361,6 +362,7 @@
   ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
   COMPONENT libraries
 )
+endif()
 
 if (BUILD_STATIC_LIB)
   add_library(${LIBSSH_STATIC_LIBRARY} STATIC ${libssh_SRCS})
