--- a/include/libssh/wrapper.h
+++ b/include/libssh/wrapper.h
@@ -52,7 +52,7 @@
 };
 
 typedef struct ssh_mac_ctx_struct *ssh_mac_ctx;
-MD5CTX md5_init(void);
+MD5CTX ssh_md5_init(void);
 void md5_update(MD5CTX c, const void *data, unsigned long len);
 void md5_final(unsigned char *md,MD5CTX c);
 
--- a/src/dh.c
+++ b/src/dh.c
@@ -918,7 +918,7 @@
     return SSH_ERROR;
   }
 
-  ctx = md5_init();
+  ctx = ssh_md5_init();
   if (ctx == NULL) {
     SAFE_FREE(h);
     return SSH_ERROR;
@@ -1049,7 +1049,7 @@
                 goto out;
             }
 
-            ctx = md5_init();
+            ctx = ssh_md5_init();
             if (ctx == NULL) {
                 free(h);
                 rc = -1;
--- a/src/kex1.c
+++ b/src/kex1.c
@@ -78,7 +78,7 @@
     ssh_string hostn) {
   MD5CTX md5 = NULL;
 
-  md5 = md5_init();
+  md5 = ssh_md5_init();
   if (md5 == NULL) {
     return -1;
   }
--- a/src/libcrypto.c
+++ b/src/libcrypto.c
@@ -234,7 +234,7 @@
   SHA512(digest, len, hash);
 }
 
-MD5CTX md5_init(void) {
+MD5CTX ssh_md5_init(void) {
   MD5CTX c = malloc(sizeof(*c));
   if (c == NULL) {
     return NULL;
--- a/src/libgcrypt.c
+++ b/src/libgcrypt.c
@@ -132,7 +132,7 @@
   gcry_md_hash_buffer(GCRY_MD_SHA512, hash, digest, len);
 }
 
-MD5CTX md5_init(void) {
+MD5CTX ssh_md5_init(void) {
   MD5CTX c = NULL;
   gcry_md_open(&c, GCRY_MD_MD5, 0);
 
