--- a/cmake/Modules/DefineCompilerFlags.cmake
+++ b/cmake/Modules/DefineCompilerFlags.cmake
@@ -21,9 +21,9 @@
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
         endif (WITH_FPIC)
 
-        check_c_compiler_flag_ssp("-fstack-protector" WITH_STACK_PROTECTOR)
+        check_c_compiler_flag_ssp("-fno-stack-protector" WITH_STACK_PROTECTOR)
         if (WITH_STACK_PROTECTOR)
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector")
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-stack-protector")
         endif (WITH_STACK_PROTECTOR)
 
         if (CMAKE_BUILD_TYPE)
