--- a/src/misc.c
+++ b/src/misc.c
@@ -215,6 +215,9 @@
 
 char *ssh_get_user_home_dir(void) {
   char *szPath = NULL;
+#ifdef ANDROID
+  return strdup(getenv("HOME"));
+#else
   struct passwd pwd;
   struct passwd *pwdbuf;
   char buf[NSS_BUFLEN_PASSWD];
@@ -233,6 +236,7 @@
   }
 
   szPath = strdup(pwd.pw_dir);
+#endif
 
   return szPath;
 }
@@ -247,6 +251,17 @@
 }
 
 char *ssh_get_local_username(void) {
+#ifdef ANDROID
+    char *name = NULL;
+    struct passwd *pwd = NULL;
+    pwd = getpwuid(getuid());
+    if (pwd == NULL) {
+        return NULL;
+    }
+
+    name = strdup(pwd->pw_name);
+
+#else
     struct passwd pwd;
     struct passwd *pwdbuf;
     char buf[NSS_BUFLEN_PASSWD];
@@ -260,6 +275,7 @@
 
     name = strdup(pwd.pw_name);
 
+#endif
     if (name == NULL) {
         return NULL;
     }
