--- a/src/misc.c
+++ b/src/misc.c
@@ -215,6 +215,9 @@
 
 char *ssh_get_user_home_dir(void)
 {
+#ifdef ANDROID
+  return strdup(getenv("HOME"));
+#else
     char *szPath = NULL;
     struct passwd pwd;
     struct passwd *pwdbuf = NULL;
@@ -235,6 +238,7 @@
     szPath = strdup(pwd.pw_dir);
 
     return szPath;
+#endif
 }
 
 /* we have read access on file */
@@ -249,6 +253,17 @@
 
 char *ssh_get_local_username(void)
 {
+#ifdef ANDROID
+    char *name = NULL;
+    struct passwd *pwd = NULL;
+    pwd = getpwuid(getuid());
+    if (pwd == NULL) {
+        return NULL;
+    }
+
+    name = strdup(pwd->pw_name);
+
+#else
     struct passwd pwd;
     struct passwd *pwdbuf = NULL;
     char buf[NSS_BUFLEN_PASSWD];
@@ -262,6 +277,7 @@
 
     name = strdup(pwd.pw_name);
 
+#endif
     if (name == NULL) {
         return NULL;
     }
