--- a/include/libssh/misc.h
+++ b/include/libssh/misc.h
@@ -35,8 +35,12 @@
 int ssh_is_ipaddr(const char *str);
 
 /* macro for byte ordering */
+#if !defined(ntohll)
 uint64_t ntohll(uint64_t);
+#endif
+#if !defined(htonll)
 #define htonll(x) ntohll(x)
+#endif
 
 /* list processing */
 
--- a/src/misc.c
+++ b/src/misc.c
@@ -303,6 +303,7 @@
 
 #endif /* _WIN32 */
 
+#if !defined(ntohll)
 uint64_t ntohll(uint64_t a) {
 #ifdef WORDS_BIGENDIAN
   return a;
@@ -315,6 +316,7 @@
   return ((((uint64_t) low) << 32) | ( high));
 #endif
 }
+#endif
 
 char *ssh_lowercase(const char* str) {
   char *new, *p;
