--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -402,6 +402,9 @@ if (NOT WIN32)
     }" HAVE_UNUSED_ATTRIBUTE)
 endif()
 
+# Stop treating warnings as errors
+unset(CMAKE_REQUIRED_FLAGS)
+
 check_c_source_compiles("
 #include <string.h>
 
@@ -440,9 +443,6 @@ int main(void) {
 }" HAVE_GCC_BOUNDED_ATTRIBUTE)
 endif(OPENBSD)
 
-# Stop treating warnings as errors
-unset(CMAKE_REQUIRED_FLAGS)
-
 # Check for version script support
 file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/conftest.map" "VERS_1 {
         global: sym;

