--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -361,6 +361,9 @@
     return 0;
 }" HAVE_GCC_NARG_MACRO)
 
+# Stop treating warnings as errors
+unset(CMAKE_REQUIRED_FLAGS)
+
 check_c_source_compiles("
 #include <stdio.h>
 int main(void) {
@@ -384,9 +387,6 @@
     return 0;
 }" HAVE_GCC_BOUNDED_ATTRIBUTE)
 
-# Stop treating warnings as errors
-unset(CMAKE_REQUIRED_FLAGS)
-
 # Check for version script support
 file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/conftest.map" "VERS_1 {
         global: sym;
