--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -380,6 +380,9 @@
     }" HAVE_UNUSED_ATTRIBUTE)
 endif()
 
+# Stop treating warnings as errors
+unset(CMAKE_REQUIRED_FLAGS)
+
 check_c_source_compiles("
 #include <string.h>
 
@@ -418,9 +421,6 @@
 }" HAVE_GCC_BOUNDED_ATTRIBUTE)
 endif(OPENBSD)
 
-# Stop treating warnings as errors
-unset(CMAKE_REQUIRED_FLAGS)
-
 # Check for version script support
 file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/conftest.map" "VERS_1 {
         global: sym;
