--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,6 +42,11 @@ macro_ensure_out_of_source_build("${PROJECT_NAME} requires an out of source buil
 # Copy library files to a lib sub-directory
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
 
+# Prevent "error C4996: ..." on windows builds
+if (WIN32)
+  add_compile_options(/wd4996)
+endif()
+
 # search for libraries
 if (WITH_ZLIB)
     find_package(ZLIB REQUIRED)

